src/Aspire.Hosting.Qdrant/QdrantServerResource.cs

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// A resource that represents a Qdrant database.
+/// </summary>
+public class QdrantServerResource : ContainerResource, IResourceWithConnectionString
+{
+    internal const string PrimaryEndpointName = "http";
+    internal const string RestEndpointName = "rest";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QdrantServerResource"/> class.
+    /// </summary>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="apiKey">A <see cref="ParameterResource"/> that contains the API Key</param>
+    public QdrantServerResource(string name, ParameterResource apiKey) : base(name)
+    {
+        ArgumentNullException.ThrowIfNull(apiKey);
+        ApiKeyParameter = apiKey;
+    }
+
+    private EndpointReference? _primaryEndpoint;
+
+    /// <summary>
+    /// Gets the parameter that contains the Qdrant API key.
+    /// </summary>
+    public ParameterResource ApiKeyParameter { get; }
+
+    /// <summary>
+    /// Gets the primary endpoint for the Qdrant database.
+    /// </summary>
+    public EndpointReference PrimaryEndpoint => _primaryEndpoint ??= new(this, PrimaryEndpointName);
+
+    /// <summary>
+    /// Gets the connection string expression for the Qdrant gRPC endpoint.
+    /// </summary>
+    public ReferenceExpression ConnectionStringExpression =>
+       ReferenceExpression.Create(
+            $"Endpoint={PrimaryEndpoint.Property(EndpointProperty.Scheme)}://{PrimaryEndpoint.Property(EndpointProperty.Host)}:{PrimaryEndpoint.Property(EndpointProperty.Port)};Key={ApiKeyParameter}");
+
+    /// <summary>
+    /// Gets the connection string expression for the Qdrant REST endpoint.
+    /// </summary>
+    public ReferenceExpression RestConnectionStringExpression =>
+        ReferenceExpression.Create(
+            $"Endpoint={PrimaryEndpoint.Property(EndpointProperty.Scheme)}://{PrimaryEndpoint.Property(EndpointProperty.Host)}:6333;Key={ApiKeyParameter}");
```

Comment:
Why is this port hard coded?

Reply:
Converting this feedback to a bug since this PR merged FYI @eerhardt

Reply:
Resolved this with https://github.com/dotnet/aspire/pull/3422